src/fundus/scraping/pipeline.py

```diff
@@ -1,47 +1,163 @@
+
+    def __call__(self) -> float:
+        """Yields back a float to specify crawler delay for current batch in seconds
+
+        This inludes the execuion time between batches.
+        The crawler will wait max(execution_time, delay)
+
+        Examples:
+            >>> import random
+            >>> delay: Delay = lambda: random.randint(1, 1000)/1000.
+            This will wait a random amount of milliseconds between batches.
+
+        Returns:
+
+        """
+        ...
+
+
 class Pipeline:
     def __init__(self, *scrapers: Scraper):
+        """Basic pipeline to utilize scrapers.
+
+        Because scrapers are implemented asynchronous this pipeline handles
+        the necessary event loops and program logic to download articles
+        in batches asynchronously.
+
+        Args:
+            *scrapers (Scraper): The scrapers which should be used.
+        """
         self.scrapers: Tuple[Scraper, ...] = scrapers
 
     def run(
         self,
         error_handling: Literal["suppress", "catch", "raise"],
         max_articles: Optional[int] = None,
         extraction_filter: Optional[ExtractionFilter] = None,
-        batch_size: int = 10,
+        delay: Optional[Delay] = lambda: 0.1,
+        url_filter: Optional[URLFilter] = None,
+        only_unique: bool = True,
     ) -> Iterator[Article]:
-        scrape_map = map(
-            lambda x: x.scrape(
-                error_handling=error_handling, batch_size=batch_size, extraction_filter=extraction_filter
-            ),
-            self.scrapers,
-        )
-        robin = more_itertools.interleave_longest(*tuple(scrape_map))
-
-        if max_articles:
-            while max_articles:
-                try:
-                    yield next(robin)
-                except StopIteration:
-                    pass
-                max_articles -= 1
+        """Yields articles from initialized scrapers.
+
+        The articles will be requested concurrently with batch size := len(self.scrapers).
+        You can specify a Delay to be used between the batches with <delay>
+
+        Works like a light-wight version of Crawler.crawl(). Ment to be used when dealing with
+        custom scrapers outside the context of predefined publisher collections.
+        Refer to the docstring of Crawler.crawl() for more detailed information about the Args.
+
+        Args:
+            max_articles (Optional[int]): Maximal number of articles to be yielded. Defaults to None.
+            error_handling (Literal["suppress", "catch", "raise"]): Set error handling
+                for extraction. Defaults to "suppress".
+            extraction_filter (Optional[ExtractionFilter]): Set extraction filter. Defaults to None.
+            delay (Optional[Delay]): Set waiting time between article batches. Defaults to None.
+            url_filter (Optional[URLFilter]): Set URLFilter. Defaults to None
+            only_unique: (bool): If true return only unique responses. Defaults to True.
```

Comment:
This is detailed in `Crawler.crawl()` as pointed out in the doc string. Judging by your comment crucial part of the doc string may be not visible enough. Any idea how to change this?

Reply:
Not really. That's on me for not reading the upper part of the docstring properly. It is unfortunate to have so many duplicated arguments to document, but that is how it is. Since the `.crawl` entry-point is the main entry point for the user, I think it is fine as you did.